### ISSUE TITLE
Magisk support or getting safetynet to pass

### ISSUE THREAD

**Issue by ShapeShifter499** [waydroid/waydroid#182]
I'm in way over my head with this sort of issue but I'd like to see some discussion in getting Magisk support working or at least getting safetynet to pass. As either one would solve the issue potentially. Right now because safetynet does not pass on a regular install of waydroid, applications that look for safetynet just do not work. 

Either waydroid could implement something to help with that problem.

Or support for Magisk could be figured out and implemented as a user opt-in option since that project is mainly about getting safetynet to pass on android when parts of the system are modified causing it to normally fail.

**Comment by teppyboy** [comment]
I don't think we will get SafetyNet to pass anytime soon, because the official Magisk has removed MagiskHide completely (if you install Magisk your device will fail the Basic Integrity), and this is an Android-x86 device, not a Google certified one so probably we can't. Still if you want to install Magisk then manually modify the Android image and add Magisk to it, should work (like some Magisk Anbox AUR packages)

**Comment by ytrezq** [comment]
@teppyboy : my bank app is neverless still working through Magiskhide despite it receive recent updates. So getting the unmainted MagiskHide support is still usefull.

Well, in my case, I sent my phone for repairs, but I’m unable to confirm payment through the bank app on waydroid in order to get it sent back to me.

**Comment by baptx** [comment]
@teppyboy @ytrezq I read Zygisk is the new MagiskHide so we may not need MagiskHide?
Here is what I tried so far:
https://github.com/casualsnek/waydroid_script/pull/143#issuecomment-1893679352

**Comment by ytrezq** [comment]
@baptx : it’s a completely different poject not a replacement as Magiskhide was removed. It doesn’t support my bank app unlike Magiskhide. It nevertheless relies on Magisk which don’t work on waydroid…

**Comment by baptx** [comment]
@ytrezq I got Magisk and MagiskHide working on Waydroid: https://github.com/casualsnek/waydroid_script/pull/143#issuecomment-1902083716
I also tried other Magisk modules but did not pass SafetyNet / Play Integrity yet: https://github.com/kdrag0n/safetynet-fix/issues/281#issuecomment-1902163729

**Comment by wandy-dev** [comment]
I gave this a go using waydroid_script and https://github.com/kdrag0n/safetynet-fix to no avail. I'd love to see this working as without waydroid doesn't meet my personal requirements. Just giving this a bump is all.

**Comment by baptx** [comment]
@Weroxig Yes, Frida worked on Waydroid, like on Anbox. I used frida-server for Android x86-64.

**Comment by ShapeShifter499** [comment]
@baptx I'm not sure but how does Frida help with safetynet?

**Comment by baptx** [comment]
@ShapeShifter499 I don't think it helps, I just wrote that I noticed a bug with Magisk.

**Comment by ayasa520** [comment]
<img width="600"  alt="Image" src="https://github.com/user-attachments/assets/fe4d0eb8-2b8a-4d63-a6bc-7e6b46156f8c" />

Please check this: https://t.me/WayDroid/225788

But Playstrong isn’t open source, which some people might mind.

**Comment by Rovyou** [comment]
@ayasa520  is still works?
